railties/lib/rails/commands/notes/notes_command.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "rails/source_annotation_extractor"
+
+module Rails
+  module Command
+    class NotesCommand < Base # :nodoc:
+      class_option :annotations, aliases: "-a", desc: "Filter by specific annotations, e.g. Foobar TODO", type: :array, default: %w(OPTIMIZE FIXME TODO)
+
+      def perform(*)
+        require_application_and_environment!
+
+        display_deprecation_warning
+        display_annotations
+      end
+
+      private
+        def display_annotations
+          annotations = options[:annotations]
+          tag = (annotations.length > 1)
+
+          Rails::SourceAnnotationExtractor.enumerate annotations.join("|"), tag: tag, dirs: directories
+        end
+
+        def directories
+          Rails::SourceAnnotationExtractor::Annotation.directories + source_annotation_directories
+        end
+
+        def display_deprecation_warning
+          return if source_annotation_directories.empty?
+          say ActiveSupport::Deprecation.warn("`SOURCE_ANNOTATION_DIRECTORIES` will be deprecated in Rails 6.1. You can add default directories by using config.annotations.register_directories instead.")
```

Comment:
Out of curiosity why is the `say` necessary here? Shouldn't we leave the Deprecation behavior decide what to do when a deprecation is triggered?

Reply:
True, that's probably not needed. Let's remove it, then I'll merge 👍

Reply:
If I get rid of the `say` here, it doesn't output in the console which I don't think is what we want.

Reply:
The user of the app is up to decide whether he wants to output deprecation. (By default it's gonna [log](https://github.com/rails/rails/blob/90e2739d8680878b40224d68b366917b9c582ba5/activesupport/lib/active_support/deprecation/behaviors.rb#L25) in the console, but we shouldn't bypass the setting)

Reply:
ohhh interesting. TIL. ok I'll make the change then.